protocol.go

```diff
@@ -85,7 +85,7 @@ type protocolHandler interface {
 	SetTimeout(*http.Request) (context.Context, context.CancelFunc, error)
 
 	// NewConn constructs a HandlerConn for the message exchange.
-	NewConn(http.ResponseWriter, *http.Request) (handlerConnCloser, bool)
+	NewConn(http.ResponseWriter, *http.Request, string) (handlerConnCloser, bool)
```

Comment:
Updated this internal interface to pass the parsed media type (without params) down to the protocol handlers. This way we can re-use the parsed data, strip out parameters in the response content type, and properly look up the codec. Without this we'd have to repeat the parsing in the protocol implementations and they don't provide a handy way to return an error.

Reply:
Rather than introducing a new parameter here, why not overwrite the request's content-type header with the canonicalized value?